kvirt/config.py

```diff
@@ -655,13 +656,23 @@ def create_vm(self, name, profile, overrides={}, customprofile={}, k=None,
                         os.makedirs(destdir, exist_ok=True)
                     common.fetch("%s/%s" % (onfly, script), destdir)
                 script = os.path.expanduser(script)
-                if basedir != '.':
+                if basedir != '.' and not script.endswith('join.sh'):
                     script = '%s/%s' % (basedir, script)
+                else:
+                    # To support defining cmds in VM rules, in KCLI plan files, when deploying K3s.
+                    # When deploying K3s the join.sh script is generated on the fly
+                    # & contains K3s cluster join info.
+                    # This in comparison to using the `scripts` claus in the workers.yml file
+                    #if script.endswith('join.sh'):
+                    script = os.path.expanduser("~/.kcli/clusters/%s/%s" % (cluster, scriptname))
+                
+                # Final conditionals before templating the lines of the script into cloud-init runcmd 
                 if script.endswith('register.sh') and skip_rhnregister_script:
                     continue
                 elif not os.path.exists(script):
                     return {'result': 'failure', 'reason': "Script %s not found" % script}
                 else:
+                    # Templating the scriptlines into cloud-init runcmd values
```

Comment:
you added the comment just to clarify the purpose of this section?

Reply:
Indeed. I felt that some comments would make wonders in that section. That's all.